### PR TITLE
Arcane Spellcasting Update

### PR DESCRIPTION
**Pull request by tioluko**
## About The Pull Request

**There is more 3 notable changes this brings:**

### 1. **New spell icons** 
(I am far from being the ideal spriter for this, but I believe I managed to make it feel much more harmonious with the rest of the UI overall) 

af

### 2. **New spells for arcane casters**   
A few of them got snatched from divine casters and others are completely new to the game (using leftover ss13 codes to guide me, and with what I am learning I feel that will be able to make more complex skills soon)

Here's the current list:

**Lightningbolt**	-Standard
**Fireball**		-Standard
**Greater Fireball**	-Standard (Only CourtWiz has access to it)
**Fetch**			-Standard
**Blindness**		-Arcane version of the existing one
**Invisibility**		-Arcane version of the existing one
**Ignite**			-Arcane version of the sacred flame
**_[New]_ Blink**	-Teleport to an selected open turn in range
**_[New]_ Swap**	-Swap places with a living target in range
**_[New]_ Smokescreen**	-Pops a small temporary smokescreen in the target area
**_[New]_ Telepathy**	-Mentally message someone in sight (not 1000% sure if the receiver gets the message correctly)
[Example Swap/Smoke gifs](https://discord.com/channels/1272201181871210597/1277293703786004632/1278819590494879956)
[Telepathy](https://discord.com/channels/1272201181871210597/1273813924671787071/1279662636577062942)

### 3. **Randomized spell selection on spawn**   
For all classes that have access to Arcane magic (**Mage, Sorcerer and Court Wizard**), I have set in a way so they always start with from the table above: **A single offensive spell**(Fireball or Lightning) and **Support/Utility spells for the rest of the selection** (keeping the total amount of spells the same as it currently is), and of course, the **Court Wizard** will always start with his ol'reliable **GreatFireball**

## Why It's Good For The Game

In my opinion, the current state for the arcane magic system on this game is **BORING**, **one dimensional** and **repetitive**
... you pick a mage and all you get fireball, lightning and a little fetch...  **OR** you get Court wizard to get all that PLUS a bigger fireball...

Of course that this is not for me alone to decide but personally I think it can be the beginning of many future small additions that will make Subterra's magic system waaaay more interesting, engaging and definitely not repetitive.

There are also some not so visible changes that are more to make it easier to expand the spell casting system in an organized manner (honestly... this codebase is far from organized the skills/spells are all over the place)
For starters the Arcane Spells are now their own magic subtype `/obj/effect/proc_holder/spell/arcane` being mostly an alternate version of the current `/obj/effect/proc_holder/spell/invoked`, I did it mostly for organization purposes, and also to avoid frustration when dealing with spells that might have a similar version on other spell casting methods (like divine).
Of course I did not touch any of the spells that are not of the Arcane type, but It can definitely make it easier to organize and differentiate all the skill/spell types later on.

I only did it because I genuinely fell in love with this game and I fell that there is a LOT of potential to untap here _(cursed spaghetti codebase aside)_ and I would love to keep contributing/trying to improve the spell casting system if you give me the green light to do so, and I have plenty of ideas not only for Arcane magic, but for [Divine casters too](https://docs.google.com/document/d/1_8Q3aNgoXhcqenR1Ez8mYPx3EIVJmRS82KSgTX2a4Ag/edit?usp=sharing)

YET, I totally understand if this ends up being rejected... at least I had fun trying xD

## Notes

I tried balance their casting cost/time/cooldowns somewhat, but be free to adjust/balance everything any way you like

The old spell list is still on spells/roguetown/wizard.dm, it is not only a backup because part of the Vampire spell list is still there (Maybe it would be better to put those in a specific file/folder?) 

Line 59 in arcane.dm is an little addition that makes so you get some Arcane Skill exp gain on cast, seems to work so far (Delete/Edit it at your leisure)

I had to do a little personal touch to Archivist and add give him +1 Arcane skill because he does start with the Fetch spell for some reason (I dont think it will be a problem even if we add the spellbook system later because there is a hard Arcane skill req for it, and 1 is far from enough)

**Known issues:**  

- [EDIT] There is a "recruit" icon that are supposed to be for jobs with recruiting capabilities (like King, Lady, Priest, etc), I tried to find and set it to all of them, but there is a chance I might have forgotten a few because I don't think I know all of them and those seem to be located in weird places 
- Could not change the Undead eyes icon as well (and they seem kinda annoying to do so in a satisfying manner atm)
- Testing it alone is far from ideal so its very understandable to be not trusting of this fork xD (testing a little to find my possible fuck ups would be nice)

### CODE REVIEW

File: code/modules/spells/roguetown/arcane_spells.dm
Comment (CPhantasm) [inline code/modules/spells/roguetown/arcane_spells.dm:1]: Could we perhaps have these broken out into respective files?
If we're looking to expand the spell roster, or add complexity to spells in the future, having them more organized than just a single file holding all of them would be nice.

Comment (tioluko) [reply code/modules/spells/roguetown/arcane_spells.dm:1]: Sure, I imagined that would eventually be needed anyways xD
Tried to sort them nicely into a folder.

**Comment by tioluko** [comment]
I also have a working branch using this one as a base for adding spellbooks that can teach those spells, and means of crafting/spawning them.   

Dont wanna shove it in there since I don't even know if this base branch is acceptable (and it's already big as it is, lets go one thing at time)

**Comment by kaylexis** [comment]
this is peak..

**Review by tioluko — COMMENTED** [review]
Added the icon_overlay for some skill/spells missing them (mostly the recruiting abilities), organizanized the spell list a bit more and did some other minor touches.

**Comment by tioluko** [comment]
I legit didn't knew renaming the base thing would close it here e.e  Sorry

**Comment by tioluko** [comment]
Just fixed a issue with the Blink spell (being able to teleport even into places you dont have line of sight is too much)

**Pull request merged**